catalog/core/catalog-core-solrcommands/src/main/java/org/codice/ddf/commands/solr/ReindexCommand.java

```diff
@@ -0,0 +1,492 @@
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardCreationException;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.source.solr.DynamicSchemaResolver;
+import ddf.catalog.source.solr.SolrMetacardClientImpl;
+import ddf.security.Subject;
+import java.io.IOException;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Option;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrQuery.ORDER;
+import org.apache.solr.client.solrj.SolrQuery.SortClause;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.solr.common.params.CursorMarkParams;
+import org.codice.ddf.security.common.Security;
+import org.codice.solr.client.solrj.SolrClient;
+import org.codice.solr.factory.SolrClientFactory;
+
+@Service
+@Command(
+  scope = SolrCommands.NAMESPACE,
+  name = "reindex",
+  description =
+      "Reindexes data from a source collection to current Catalog Framework collection(s)."
+)
+public class ReindexCommand extends SolrCommands {
+  private static final int PAGE_SIZE = 200;
+
+  private static final int WRITE_TXN_SIZE = 100;
+
+  @VisibleForTesting protected static final String EARLY_TIME = "1900-01-01T00:00:00.000Z";
+
+  private Security security = Security.getInstance();
+
+  private String cursorMark = CursorMarkParams.CURSOR_MARK_START;
+
+  private Reader readerThread = null;
+
+  private SolrMetacardClientImpl metacardClient =
+      new SolrMetacardClientImpl(null, null, null, new DynamicSchemaResolver());
+
+  private long totalCount = 0;
+
+  private AtomicLong count = new AtomicLong();
+
+  private long startTime;
+
+  private ThreadPoolExecutor publishExecutor;
+
+  private SolrClient solrjClient = null;
+
+  @Reference private CatalogFramework catalogFramework;
+
+  @Reference private SolrClientFactory clientFactory;
+
+  @Option(
+    name = "-s",
+    aliases = {"--source"},
```

Comment:
:pencil2: Source is an overloaded term in relation to the catalog.  Maybe `--solr` instead.